internal/experimental/experimental.go

```diff
@@ -0,0 +1,77 @@
+package experimental
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	envName = "NOTATION_EXPERIMENTAL"
+	enabled = "1"
+)
+
+// IsDisabled determines whether an experimental feature is disabled.
+func IsDisabled() bool {
+	return os.Getenv(envName) != enabled
+}
+
+// Error returns an error for a disabled experimental feature.
+func Error(description string) error {
+	return fmt.Errorf("%q is marked as experimental and not enabled by default. To use it, please set %s=%s in your environment", description, envName, enabled)
+}
+
+// CheckCommandAndWarn checks whether an experimental command can be run.
+func CheckCommandAndWarn(cmd *cobra.Command, args []string) error {
+	if err := Check(func() (string, bool) {
+		return cmd.CommandPath(), true
+	}); err != nil {
+		return err
+	}
+	return Warn()
+}
+
+// FlagsChecker returns the function which checks whether flags are not set.
+func FlagsChecker(flags ...string) func(cmd *cobra.Command, args []string) error {
+	return func(cmd *cobra.Command, args []string) error {
+		if err := Check(func() (string, bool) {
+			var changed []string
+			for _, flag := range flags {
+				fs := cmd.Flags()
```

Comment:
```suggestion
				flagSet := cmd.Flags()
```
usually, `fs` refers to package `io/fs` across notation.